### PR TITLE
[Blazor] DynamicComponents - fixes samples vulnerability + related adjustments

### PR DESCRIPTION
**Pull request by hakenr**
@guardrex, this draws the new samples from https://github.com/dotnet/blazor-samples/pull/256 and includes related adjustments

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [aspnetcore/blazor/components/dynamiccomponent.md](https://github.com/dotnet/AspNetCore.Docs/blob/0538fadd32792fb47d0ede5ab9f86aac6e96eb5a/aspnetcore/blazor/components/dynamiccomponent.md) | [Dynamically-rendered ASP.NET Core Razor components](https://review.learn.microsoft.com/en-us/aspnet/core/blazor/components/dynamiccomponent?branch=pr-en-us-32251) |

### CODE REVIEW

File: aspnetcore/blazor/components/dynamiccomponent.md [hunk 1]
```diff
@@ -143,8 +143,10 @@ In the following example, a Razor component renders a component based on the use
 
 In the preceding example:
 
-* Component names are used as the option values using the [`nameof` operator](/dotnet/csharp/language-reference/operators/nameof), which returns component names as constant strings.
-* The namespace of the app is `BlazorSample`. ***Change the namespace to match your app's namespace.***
+* A `Dictionary<string, Type>` is used to manage components to be displayed.
+* Names serve as the dictionary keys (and are provided as selection options).
```
Comment (guardrex) [inline aspnetcore/blazor/components/dynamiccomponent.md:147]:
```suggestion
* Names serve as the dictionary keys and are provided as selection options.
```

**Review by guardrex — CHANGES_REQUESTED** [review]
Thanks for all of this work, @hakenr!

I just left a few super NIT suggestions.

**Review by guardrex — APPROVED** [review]
# 🎉

**Comment by guardrex** [comment]
BTW @hakenr ... We're using a tool that Kirk Larkin (@serpent5) made to efficiently get XREF links. For API like `Dictionary<TKey, TValue>`, where there's some encoding and numbering, it's very helpful. It taps the same service the the API browser uses ...

https://rs-xrefs.azurewebsites.net/

**Comment by hakenr** [comment]
> BTW @hakenr ... We're using a tool that Kirk Larkin (@serpent5) made to efficiently get XREF links. For API like `Dictionary<TKey, TValue>`, where there's some encoding, special characters, and numbering, it's very helpful. It taps the same service that the API browser uses ...
> 
> https://rs-xrefs.azurewebsites.net/

BTW, my first (local) version was using the XREF to `Dictionary<TKey, TValue>`, but in the end I decided to use the `Dictionary<string, Type>` version in the text and thus dropped the XREF (is there a way to combine the two?).

But don't worry, the generic `Dictionary<TKey, TValue>` works here, no need to use type-specific version in the text.

For me, another concern with using the XREF is deciding whether to sacrifice the `code` style in favor of a [link](#). (In our component library, we combine these into [`CodeWithLink`](#), but it is obviously not used in MS docs.

**Comment by guardrex** [comment]
Indeed, I've had my own tiny pet peeve about the styling for many years.

Anyway ... yes ... combine the XREF with code-fenced link text ...

> \[\`Dictionary<string, Type>\`](xref:System.Collections.Generic.Dictionary%602)

That format would be a bit atypical from a C# Guide perspective. I don't think they ever do that, but we sometimes do :smile:. I'm sure if Bill Wagner saw some of the wild stuff over here, he'd roll his eyes and curse our bones! :laughing: I've used atypical link text in some cases. I doubt anyone is very confused or upset by it. I think the API browser cross-links are valuable in that case, and it did occur to me when I floated the suggestion to do that. However, I think just cross-linking the dictionary is best. Readers should probably get used to seeing `TKey`/`TValue` things, as it's everywhere in the API browser and C# Guide.

I would like to get more of the types without cross-links addressed. Sometimes, I'm just rushing and don't place them. Sometimes, new API can't be cross-linked because we're in a preview. Sometimes, the PU ... ours and other PUs ... fail to expose them in the public API. We must put in a request to have them surfaced, and then I hopefully have a tracking item to circle around and fix up the text later.

**Comment by hakenr** [comment]
👍

**Pull request merged**